src/components/ec/cuda/kernel/ec_cuda_executor.cu

```diff
@@ -290,6 +172,151 @@ __device__ void executor_copy_multi(ucc_eee_task_copy_multi_t *task)
+                         align_pow2((intptr_t)task->reduce.srcs[0], 16) ||
+                         align_pow2((intptr_t)task->reduce.srcs[1], 16));
+    } else {
+        ucc_assert(task->task_type == UCC_EE_EXECUTOR_TASK_REDUCE_STRIDED);
+        dt      = task->reduce_strided.dt;
+        count   = task->reduce_strided.count;
+        op      = task->reduce_strided.op;
+        n_src   = task->reduce_strided.n_src2 + 1;
+        s1      = task->reduce_strided.src1;
+        s2      = task->reduce_strided.src2;
+        d       = task->reduce_strided.dst;
+        aligned = !(align_pow2((intptr_t)task->reduce_strided.dst, 16) ||
+                     align_pow2((intptr_t)task->reduce_strided.src1, 16) ||
+                     align_pow2((intptr_t)task->reduce_strided.src2, 16));
+    }
+
+    if (count == 0) {
+        return UCC_OK;
+    }
+
+    if (UCC_DT_FLOAT32 == dt && UCC_OP_SUM == op && aligned && n_src == 2) {
+            executor_reduce_float_sum_aligned_2((float *)s1, (float *)s2,
+                                                (float *)d, count);
+            return UCC_OK;
+    }
+    switch (dt) {
+    case UCC_DT_INT8:
+        DT_REDUCE_INT(int8_t, task, op);
+        break;
+    case UCC_DT_INT16:
+        DT_REDUCE_INT(int16_t, task, op);
+        break;
+    case UCC_DT_INT32:
+        DT_REDUCE_INT(int32_t, task, op);
+        break;
+    case UCC_DT_INT64:
+        DT_REDUCE_INT(int64_t, task, op);
+        break;
+    case UCC_DT_UINT8:
+        DT_REDUCE_INT(uint8_t, task, op);
+        break;
+    case UCC_DT_UINT16:
+        DT_REDUCE_INT(uint16_t, task, op);
+        break;
+    case UCC_DT_UINT32:
+        DT_REDUCE_INT(uint32_t, task, op);
+        break;
+    case UCC_DT_UINT64:
+        DT_REDUCE_INT(uint64_t, task, op);
+        break;
+    case UCC_DT_FLOAT16:
+        DT_REDUCE_FLOAT(__half, task, op);
+        break;
+    case UCC_DT_FLOAT32:
+#if SIZEOF_FLOAT == 4
+        DT_REDUCE_FLOAT(float, task, op);
+        break;
+#else
+        return UCC_ERR_NOT_SUPPORTED;
+#endif
+    case UCC_DT_FLOAT64:
+#if SIZEOF_DOUBLE == 8
+        DT_REDUCE_FLOAT(double, task, op);
+        break;
+#else
+        return UCC_ERR_NOT_SUPPORTED;
+#endif
+    case UCC_DT_FLOAT32_COMPLEX:
+#if SIZEOF_CUFLOATCOMPLEX == 8
+        DT_REDUCE_FLOAT_COMPLEX(cuFloatComplex, float, task, op);
+        break;
+#else
+        return UCC_ERR_NOT_SUPPORTED;
+#endif
+    case UCC_DT_FLOAT64_COMPLEX:
+#if SIZEOF_CUDOUBLECOMPLEX == 16
+        DT_REDUCE_FLOAT_COMPLEX(cuDoubleComplex, double, task, op);
+        break;
+#else
+        return UCC_ERR_NOT_SUPPORTED;
+#endif
+#if CUDART_VERSION >= 11000
+    case UCC_DT_BFLOAT16:
+        ucc_assert(2 == sizeof(__nv_bfloat16));
+        DT_REDUCE_FLOAT(__nv_bfloat16, task, op);
+        break;
+#endif
+    default:
+        return UCC_ERR_NOT_SUPPORTED;
+    }
+    return UCC_OK;
+}
+
+__device__ void executor_reduce_multi_dst_task(ucc_eee_task_reduce_multi_dst_t *task)
```

Comment:
Done.
Should I also remove `executor_copy_multi` defined in `ec/rocm/kernel/ec_rocm_executor_kernel.cu` ? From what I understand we should keep it since `ucc_rocm_executor_persistent_task_post` does not handle this case.